client/src/components/AppContext.tsx

```diff
@@ -0,0 +1,18 @@
+import { createContext } from 'react';
+import { Auth, User } from '../lib';
+
+type AppContextValues = {
+  user: User | undefined;
+  token: string | undefined;
+  handleSignIn: (auth: Auth) => void;
+  handleSignOut: () => void;
+};
+
+const AppContext = createContext<AppContextValues>({
+  user: undefined,
+  token: undefined,
+  handleSignIn: () => undefined,
+  handleSignOut: () => undefined,
+});
```

Comment:
It's a nuisance having to provide a dummy context value, isn't it? But this is the right thing to do and is essentially what I've done many times.

I've tried just using `null` as an initial context value since `createContext` defaults to null. However, then my `useContext` callers all have to deal with a null value, which seems worse than the default value like you've done.